compiler/test/dotty/tools/dotc/reporting/CodeActionTest.scala

```diff
@@ -136,6 +136,24 @@ class CodeActionTest extends DottyTest:
          afterPhase = "patternMatcher"
       )
 
+  @Test def removeUnusedLocalDefinition =
+    checkCodeAction(
+      code = """object Test:
+         |  def foo(): Int = {
+         |    val a = 456
+         |    123
+         |  }
+         |""".stripMargin,
+      title = "Remove unused code",
+      expected = """object Test:
+         |  def foo(): Int = {
+         |    123
+         |  }
+         |""".stripMargin , 
+         afterPhase = "checkUnusedPostInlining"
```

Comment:
**thought:** In theory there are also other cases with variable definitions e.g.:
```scala
val List(a, b) = List(1, 2)
```

but they are not being detected at the moment